### PR TITLE
v0.17.0

### PR DESCRIPTION
**Pull request by amilajack**
- [x] upgraded to `babel@7` (thanks to @vikr01 🎉🎉🎉)
- [x] migrated from [two `package.json` structure](https://www.electron.build/tutorials/two-package-structure) (thanks to @HyperSprite!)
- [x] initial auto update support (experimental)
- [x] migrate from greenkeeper to [renovate](https://renovatebot.com)
- [x] added issue template
- [x] use `babel-preset-env` to target current electron version
- [x] add [opencollective](https://opencollective.com/electron-react-boilerplate-594) banner message display in postinstall script (help support ERB 🙏)
- [x] fix failing ci issues
- [x] add migration guide

### CODE REVIEW

File: configs/webpack.config.base.js [hunk 1]
```diff
@@ -62,7 +35,7 @@ export default {
    */
   resolve: {
     extensions: ['.js', '.jsx', '.json'],
-    modules: [path.join(__dirname, '..', 'app'), 'node_modules']
+    modules: [path.join(__dirname, '..'), 'node_modules']
```
Comment (vikr01) [inline configs/webpack.config.base.js]: I think we can just remove this whole line. I believe it was originally meant to be `path.join(__dirname, '..', 'app/node_modules')`. Was changed during the single package.json setup in #1854.

**Comment by amilajack** [comment]
>  I think we should add a note of how to work with these changes to the

Good idea. I think the `CHANGELOG.md` should strictly be a list of changes. IMO we should move the migration steps to a migration guide. [Laravel does this nicely with their upgrade guide](https://laravel.com/docs/5.7/upgrade):

screen shot 2018-10-31 at 12 40 24 pm

**Comment by amilajack** [comment]
[Added upgrade guide here](https://electron-react-boilerplate.js.org/docs/en/next/upgrade-guide)

**Comment by dbvrac** [comment]
It would help me A LOT if I could get an annotated visual representation of the upgrade tree. I'm still working with v0.14.0. It's hard to develop my skills, my app, AND keep up with the upgrades.

**Comment by amilajack** [comment]
@dbvrac. I'd love to help but I'm not sure what you mean by "visual representation of the upgrade tree". Can you please expand on that.

**Comment by dbvrac** [comment]
@amilajack Some kind of graph, like this one from [https://neo4j.com/blog/scribestar-neo4j-faster-coding-cycles/](https://neo4j.com/blog/scribestar-neo4j-faster-coding-cycles/)
scribestar-versioning-data-model-tree
*sorry about the potato quality*

From my POV it seems like a significant undertaking and I have no time to put into it. However, considering the pool of talent on this project maybe someone else could whip it together in a jiffy. I think it would really help new users negotiate the learning curve.

**Comment by dbvrac** [comment]
Not exactly what I had in mind, but FWIW here's [this video](https://youtu.be/4LHTaWo6DTc) of the evolution of the ERB project, brought to you by [gource](https://gource.io/).

Screenshot: 
erb-gource

**Comment by talha131** [comment]
@amilajack is the package size issue as reported here https://github.com/electron-react-boilerplate/electron-react-boilerplate/pull/1986 have been fixed? I would greatly appreciate if you can guide about this before I upgrade to 0.17.1 release.

**Comment by amilajack** [comment]
That has not been fixed yet. I would recommend waiting until we fix that before upgrading.

**Pull request merged**